test/json-checker.jl

```diff
@@ -0,0 +1,62 @@
+# Run JSON checker tests
+
+const JSON_DATA_DIR = joinpath(dirname(dirname(@__FILE__)), "data")
+const ALLOWED_CHECK_FAILURES = [1, 2, 7, 8, 10, 18, 25, 27, 32]
+const ALLOWED_RT_FAILURES = [17, 18, 19]
+
+for i in 1:33
+    file = "fail$(lpad(i, 2, 0)).json"
+    filepath = joinpath(JSON_DATA_DIR, "jsonchecker", file)
+
+    # FIXME: This is a convoluted way of allowing certain tests to fail. Once
+    # ALLOWED_CHECK_FAILURES is empty, we can eliminate the extra code.
+
+    if i ∈ ALLOWED_CHECK_FAILURES
+        try
+            JSON.parsefile(filepath)
+
+            print_with_color(:red, " [Fail] ")
+            println("Test $file parsed successfully; should throw ErrorException.")
+            println(e)
+        catch e
+            if isa(e, ErrorException)
```

Comment:
Doesn't need to be fixed right now, but I wonder if the exception types thrown for various invalid data or parsing failures should be made more specific.